src/hepattn/models/wrapper.py

```diff
@@ -0,0 +1,175 @@
+import torch
+import torch.nn as nn
+
+from lightning import LightningModule
+from torchjd import mtl_backward
+from torchjd.aggregation import UPGrad
+
+
+class ModelWrapper(LightningModule):
+    def __init__(
+            self,
+            model: nn.Module,
+            lrs_config: dict,
+            optimizer: str = "AdamW",
+            mtl: bool = False,
+        ):
+        super().__init__()
+
+        self.save_hyperparameters(logger=False)
+
+        self.model = model
+        self.optimizer = optimizer
+        self.lrs_config = lrs_config
+        self.mtl = mtl
+        
+        # If we are doing multi-task-learning, optimisation step must be done manually
+        if mtl:
+            self.automatic_optimization = False
+
+    def forward(self, inputs):
+        return self.model(inputs)
+
+    def predict(self, outputs):
+        return self.model.predict(outputs)
+
+    def log_losses(self, losses, stage):
+        for layer_name, layer_losses in losses.items():
+            for task_name, task_losses in layer_losses.items():
+                task_losses = losses[layer_name][task_name]
+                self.log_dict({f"{stage}_{layer_name}_{task_name}_{k}": v for k, v in task_losses.items()})
+
+    def log_task_metrics(self, preds, targets, stage):
+        # Log any task specific metrics
+        for task in self.model.tasks:
+            # Check that the task actually has some metrics to log
+            if not hasattr(task, "metrics"): continue
+
+            # Just log the predictions from the final layer for now
+            task_metrics = task.metrics(preds["final"][task.name], targets)
+
+            # If the task returned a non-empty metrics dict, log it
+            if task_metrics:
+                self.log_dict({f"{stage}_{task.name}_{k}": v for k, v in task_metrics.items()})
```

Comment:
```suggestion
                self.log_dict({f"{stage}/{task.name}_{k}": v for k, v in task_metrics.items()})
```
and similar elsewhere